samples/client-sdk/getting-started-daily-js/index.js

```diff
@@ -0,0 +1,251 @@
+   */
+  destroyTrack(event) {
+    let tracks = document.querySelectorAll('video, audio');
+    tracks.forEach((el) => {
+      if (el.srcObject && el.srcObject.getTracks().includes(event.track)) {
+        el.remove();
+      }
+    });
+  }
+
+  /**
+   * Updates UI for track state changes (e.g., mute states).
+   */
+  handleParticipantStateUpdate(event) {
+    this.getTrackStates();
+  }
+
+  /**
+   * Updates the UI with the current active speaker's identity.
+   * @param {Object} event - The active speaker change event object.
+   */
+  handleActiveSpeakerChange(event) {
+    document.getElementById(
+      'active-speaker'
+    ).textContent = `Active Speaker: ${event.activeSpeaker.peerId}`;
+  }
+
+  /**
+   * Tries to join a call with provided room URL and optional join token.
+   * @param {string} roomUrl - The URL of the room to join.
+   * @param {string|null} joinToken - An optional token for joining the room.
+   */
+  async joinRoom(roomUrl, joinToken = null) {
+    if (!roomUrl) {
+      console.error('Room URL is required to join a room.');
+      return;
+    }
+
+    const joinOptions = { url: roomUrl };
+    if (joinToken) {
+      joinOptions.token = joinToken;
+      console.log('Joining with token:', joinToken);
+    } else {
+      console.log('Joining without a token.');
+    }
+
+    try {
+      await this.call.join(joinOptions);
+      console.log(`Successfully joined: ${roomUrl}`);
+    } catch (e) {
+      console.error('Join failed:', e);
+    }
+  }
+
+  /**
+   * Retrieves the track states of the local participant's tracks and updates the UI.
+   */
+  getTrackStates() {
+    if (this.call && this.call.participants().local) {
+      const localParticipant = this.call.participants().local;
+      this.isCameraMuted = localParticipant.tracks.video.state === 'off';
+      this.isMicMuted = localParticipant.tracks.audio.state === 'off';
+      this.updateUiForTrackStates();
+    }
+  }
+
+  /**
+   * Toggles the local video track's mute state.
+   */
+  toggleCamera() {
+    this.call.setLocalVideo(this.isCameraMuted);
+    this.isCameraMuted = !this.isCameraMuted;
+    this.updateUiForTrackStates();
+  }
+
+  /**
+   * Toggles the local audio track's mute state.
+   */
+  toggleMicrophone() {
+    this.call.setLocalAudio(this.isMicMuted);
+    this.isMicMuted = !this.isMicMuted;
+    this.updateUiForTrackStates();
+  }
+
+  /**
+   * Updates the UI to reflect the current states of the local participant's camera and microphone.
+   */
+  updateUiForTrackStates() {
+    document.getElementById('camera-state').textContent = `Camera: ${
+      this.isCameraMuted ? 'Off' : 'On'
+    }`;
+    document.getElementById('mic-state').textContent = `Mic: ${
+      this.isMicMuted ? 'Off' : 'On'
+    }`;
+  }
+
+  /**
+   * Leaves the call and performs necessary cleanup operations like removing video elements.
+   */
+  async leave() {
+    try {
+      await this.call.leave();
```

Comment:
Wondering the same thing here  as the join handling - maybe worth using the `"left-meeting"` event here?